packages/happn-db-provider-loki/index.js

```diff
@@ -828,4 +843,16 @@ module.exports = class LokiDataProvider extends commons.BaseDataProvider {
       throw new Error('Loaded archives only support read-only operations!');
     }
   }
+
+  #getFileSize(filepath) {
+    let fileStats = null;
+    try {
+      fileStats = fs.statSync(filepath);
```

Comment:
Why are we doing this all as Sync function?  we should rather just put locks around changing the DB while we are storing and not block other async IO.  

I think calculating size, will be very depend on the file size, calculation on a couple of Megs will take a while.

Reply:
we took a quantitative look at this - it's a native Node.js operation relying on the cached file system, and measured it to take < 1ms regardless of file size. So the cost is negligible compared to real-life usage patterns and does not merit further optimization.